src/nvim/popupmnu.c

```diff
@@ -904,16 +904,30 @@ int pum_get_height(void)
   return pum_height;
 }
 
+/// Gets the internal pum geometry.
+///
+/// @return the internal pum geometry. Ignores UI external pum geometry.
+/// Only valid when pum_visible() returns TRUE!
+void pum_get_internal_pos(int *pwidth, int *pheight, int *prow, int *pcol)
+{
+  *pwidth = pum_width;
+  *pheight = pum_height;
+  *prow = pum_row;
+  *pcol = pum_col;
+}
+
 /// Add size information about the pum to "dict".
 void pum_set_event_info(dict_T *dict)
 {
   if (!pum_visible()) {
     return;
   }
-  tv_dict_add_nr(dict, S_LEN("height"), pum_height);
-  tv_dict_add_nr(dict, S_LEN("width"), pum_width);
-  tv_dict_add_nr(dict, S_LEN("row"), pum_row);
-  tv_dict_add_nr(dict, S_LEN("col"), pum_col);
+  double w, h, r, c;
+  ui_pum_get_pos(&w, &h, &r, &c);
```

Comment:
Now I think `ui_pum_get_pos` could return a `bool` if any UI was found. Then this could be 
```
if(!ui_pum_get_pos(&w, &h, &r, &c)) {
  w = pum_width;
  h = pum_height;
  ...
}
```

and `pum_get_internal_pos` removed.